src/sql-parser/src/parser.rs

```diff
@@ -6596,19 +6606,20 @@ impl<'a> Parser<'a> {
             .map_parser_err(StatementKind::Prepare)?;
         let pos = self.peek_pos();
         //
-        let stmt = match self.parse_statement()? {
-            stmt @ Statement::Select(_)
-            | stmt @ Statement::Insert(_)
-            | stmt @ Statement::Delete(_)
-            | stmt @ Statement::Update(_) => stmt,
-            _ => {
-                return parser_err!(self, pos, "unpreparable statement")
-                    .map_no_statement_parser_err()
-            }
-        };
+        let (stmt, sql) = self.parse_statement()?;
+        if !matches!(
+            stmt,
+            Statement::Select(_)
+                | Statement::Insert(_)
+                | Statement::Delete(_)
+                | Statement::Update(_),
+        ) {
+            return parser_err!(self, pos, "unpreparable statement").map_no_statement_parser_err();
+        }
```

Comment:
According to the coverage build, this code is covered by unit tests but not by higher level tests. Would it be possible to add an end-to-end test to cover this snippet / function?